### PR TITLE
allow injecting CsrfTokenManager by class

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

File: lib/private/Server.php [hunk 1]
```diff
@@ -1026,6 +1026,7 @@ public function __construct($webRoot, \OC\Config $config) {
 				$c->query(SessionStorage::class)
 			);
 		});
+		$this->registerAlias(CsrfTokenManager::class, 'CsrfTokenManager');
```
Comment (MorrisJobke) [inline lib/private/Server.php]: Why is this needed? It's the same name, or do I miss something? If the type hinting is there it should work already.

Comment (icewind1991) [reply lib/private/Server.php]: Currently when type hinting it creates an instance that is separate from when using `$server->query('CsrfTokenManager')` since the container doesn't recognize them as the same object so it will create one instance for `OC\Security\CSRF\CsrfTokenManager` and one for `CsrfTokenManager`

Comment (rullzer) [reply lib/private/Server.php]: What? This doesn't make sense it should resolve the object. Not the name.

Comment (rullzer) [reply lib/private/Server.php]: Ah ok I read wrong. You should always query the full classname

Comment (rullzer) [reply lib/private/Server.php]: Ok so the fix is indeed not wrong. But I prefer it if we alias the 'string'. So we can eventually deprecate that stuff properly.

**Comment by rullzer** [comment]
Pushed a fixup that I think if more clean.

**Review by rullzer — APPROVED** [review]

File: lib/private/Server.php [hunk 2]
```diff
@@ -1901,7 +1902,7 @@ public function getSessionCryptoWrapper() {
 	 * @return CsrfTokenManager
 	 */
 	public function getCsrfTokenManager() {
```
Comment (MorrisJobke) [inline lib/private/Server.php]: All those `get...` methods in the DI container feel sooo wrong 😭 Anyways ... in general the code looks now much better 👍

**Review by MorrisJobke — APPROVED** [review]
